firebase-common/src/test/java/com/google/firebase/platforminfo/FirebasePlatformLoggingTest.java

```diff
@@ -79,7 +80,7 @@ public void test_watch_atNotHighEnoughApiLevel() {
         app -> {
           UserAgentPublisher ua = app.get(UserAgentPublisher.class);
 
-          assertThat(ua.getUserAgent()).contains("android-platform/ ");
+          assertThat(ua.getUserAgent()).containsMatch(Pattern.compile("android-platform/($|\\s)"));
```

Comment:
Having played with this regex a bit, it does not look like what we want here. afaict it will *only* accept: `foo/` and `foo/ ` but not `foo/ bar/`.
What the test should ensure, is that if the value is not last, then there should be at least one space after it, so I suggest to use a pattern like `"android-platform/(\\s+.*)*"`. wdyt?

Reply:
Also I am curious, is master broken because of this?

Reply:
As discussed offline, the pattern does find the pattern :-) 

If not master, at least 2 other PRs: #2698 and #2684